app.js

```diff
@@ -9,11 +8,71 @@ class Dashboard extends React.Component {
     /**
      * @inheritdoc
      */
+    constructor(props) {
+        super(props);
+        this.psnrTestResultTransformFunc = jsonData => {
```

Comment:
this (defining this all here) doesn't feel great to me.  curious if you have any suggestions.

Reply:
Why are you defining it in the constructor?

Reply:
i tried as a member but was running into problems for some reason, probably just syntax mistakes on my part.  i don't think it feels great as a member either though, really.

Reply:
To be fair, I think previously you were mixing view logic with model logic.